resources/prosody-plugins/mod_room_metadata_component.lua

```diff
@@ -62,27 +65,31 @@ function getMetadataJSON(room)
     return res;
 end
 
--- Putting the information on the config form / disco-info allows us to save
--- an extra message to users who join later.
-function getFormData(room)
-    return {
-        name = FORM_KEY;
-        type = 'text-multi';
-        label = 'Room metadata';
-        value = getMetadataJSON(room);
-    };
-end
-
 function broadcastMetadata(room)
     local json_msg = getMetadataJSON(room);
 
+    if not json_msg then
+        return;
+    end
+
     for _, occupant in room:each_occupant() do
-        send_json_msg(occupant.jid, internal_room_jid_match_rewrite(room.jid), json_msg)
+        send_metadata(occupant, internal_room_jid_match_rewrite(room.jid), json_msg)
     end
 end
 
-function send_json_msg(to_jid, room_jid, json_msg)
-    local stanza = st.message({ from = module.host; to = to_jid; })
+function send_metadata(occupant, room_jid, json_msg)
+    if not json_msg or is_admin(occupant.bare_jid) then
+        local metadata_to_send = room.jitsiMetadata or {};
+
+        -- we want to send the main meeting participants only to jicofo
+        if is_admin(occupant.bare_jid) and room._data.mainMeetingParticipants then
+            metadata_to_send.mainMeetingParticipants = room._data.mainMeetingParticipants;
```

Comment:
Is this going to insert `mainMeetingParticipants` into `room.jitsiMetadata`, or is there a copy happening somewhere that I don't see?